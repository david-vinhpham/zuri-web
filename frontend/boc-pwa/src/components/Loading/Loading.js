import React, { Component } from 'react';
import { bool } from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { PulseLoader } from 'react-spinners';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import { brand04 } from 'constants/colors';
import s from './Loading.css';

class Loading extends Component {
  static propTypes = {
    loading: bool,
  };

  static defaultProps = {
    loading: null,
  };

  state = {};

  render() {
    const { loading } = this.props;
    return (
      loading && (
        <div className={s.container}>
          <PulseLoader color={brand04} size={20} sizeUnit="px" loading />
        </div>
      )
    );
  }
}

const mapStateToProps = ({ common }) => ({
  loading: common.isLoading,
});

const enhancers = [withStyles(s), connect(mapStateToProps)];
export default compose(...enhancers)(Loading);
