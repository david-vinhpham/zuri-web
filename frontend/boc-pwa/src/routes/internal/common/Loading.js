import React, { PureComponent } from 'react';
import { FullLoading, Cycle } from 'homecredit-ui';
import { bool } from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';

class Loading extends PureComponent {
  static propTypes = {
    loading: bool,
  };

  static defaultProps = {
    loading: false,
  };

  render() {
    const FullLoadingCycle = FullLoading(Cycle);
    const { loading } = this.props;
    return <div>{loading && <FullLoadingCycle fullPage />}</div>;
  }
}

function mapStateToProps({ internal }) {
  const { loading } = internal;
  return { loading };
}

const enhance = compose(connect(mapStateToProps));

export default enhance(Loading);
