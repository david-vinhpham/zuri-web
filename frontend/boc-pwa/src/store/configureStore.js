import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { persistStore, persistReducer } from 'redux-persist';

import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';

import storage from 'redux-persist/lib/storage';
import { name, version } from '../../package.json';
import rootReducer from '../reducers';
import createLogger from './logger';

export default function configureStore(initialState, helpersConfig) {
  const middleware = [thunk.withExtraArgument(helpersConfig)];

  const persistConfig = {
    key: 'BOCVN',
    storage,
    stateReconciler: autoMergeLevel2,
    blacklist: ['clz'],
  };

  const persistedReducer = persistReducer(persistConfig, rootReducer);

  let enhancer;

  if (__DEV__) {
    middleware.push(createLogger());

    // https://github.com/zalmoxisus/redux-devtools-extension#14-using-in-production
    const composeEnhancers = composeWithDevTools({
      // Options: https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md#options
      name: `${name}@${version}`,
    });

    // https://redux.js.org/docs/api/applyMiddleware.html
    enhancer = composeEnhancers(applyMiddleware(...middleware));
  } else {
    enhancer = applyMiddleware(...middleware);
  }

  // https://redux.js.org/docs/api/createStore.html
  const store = createStore(persistedReducer, enhancer);

  // Hot reload reducers (requires Webpack or Browserify HMR to be enabled)
  if (__DEV__ && module.hot) {
    module.hot.accept('../reducers', () =>
      // Don't forget to remove `()` if you change reducers back to normal rootReducer.
      // eslint-disable-next-line global-require
      store.replaceReducer(require('../reducers').default()),
    );
  }

  const persistor = persistStore(store);

  return { store, persistor };
}
