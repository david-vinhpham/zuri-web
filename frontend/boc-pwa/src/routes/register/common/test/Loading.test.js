import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import Loading from '../Loading';

const mockStore = configureStore();
const initialState = {
  saRegister: {
    loading: true,
  },
};

it('Loading snapshot', () => {
  const store = mockStore(initialState);
  const wrapper = renderer.create(
    <Provider store={store}>
      <Loading />
    </Provider>,
  );
  const snapshot = wrapper.toJSON();

  expect(snapshot).toMatchSnapshot();
});
