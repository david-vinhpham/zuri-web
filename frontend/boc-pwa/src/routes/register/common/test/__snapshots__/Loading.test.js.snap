// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading snapshot 1`] = `
<div
  className="FullLoading_wrapper__2oo9M FullLoading_wrapperFullPage__1OMTm"
>
  <svg
    className="Cycle_circular__2Unpq"
    viewBox="25 25 50 50"
  >
    <circle
      className="Cycle_path__2mEbu"
      cx="50"
      cy="50"
      fill="none"
      r="20"
      strokeMiterlimit="10"
      strokeWidth="3"
    />
  </svg>
</div>
`;
